pkg/accounting/accounting.go

```diff
@@ -126,19 +131,59 @@ func (a *Accounting) Credit(peer swarm.Address, price uint64) error {
 
 	a.logger.Tracef("crediting peer %v with price %d, new balance is %d", peer, price, nextBalance)
 
+	// compute expected debt before update because reserve still includes the amount that is deducted from the balance
+	expectedDebt := balance.expectedDebt()
+
 	err = a.store.Put(balanceKey(peer), nextBalance)
 	if err != nil {
 		return err
 	}
 
 	balance.balance = nextBalance
 
-	// TODO: try to initiate payment if payment threshold is reached
-	// if balance.balance < -int64(a.paymentThreshold) { }
+	// if our expected debt exceeds our payment threshold (which we assume is also the peers payment threshold), trigger settlement
+	if expectedDebt >= a.paymentThreshold {
+		a.settle(peer, balance)
+	}
 
 	return nil
 }
 
+// settle all debt with a peer
+// the lock on balance must be held when called
+func (a *Accounting) settle(peer swarm.Address, balance *PeerBalance) {
```

Comment:
If settle function would return an error, error messages would not have to contain function arguments, just reasons for failure, and the logging could be done where the function is called with any additional information required in its context, or to act accordingly if there is an error. One case is if we want to continue on calling settle if error rate is very high, which we cannot detect currently.

Reply:
I changed the error handling a bit. This now returns an error. I also removed the peer from all the other error messages in accounting.